From 1755ec6d1e0d09d74de815a94dfeac8b7f4d88f4 Mon Sep 17 00:00:00 2001
From: Sergei Trofimovich <slyfox@gentoo.org>
Date: Mon, 27 Feb 2012 19:42:08 +0000
Subject: fix build failures against ghc-7.4.1

$ runhaskell Setup.hs build
Building bustle-0.4.0.1...
Preprocessing executable 'bustle' for bustle-0.4.0.1...

Bustle.hs:1:1:
    Ambiguous module name `Prelude':
      it was found in multiple packages: base haskell98-2.0.0.1

Signed-off-by: Sergei Trofimovich <slyfox@gentoo.org>
---
diff --git a/Bustle/Noninteractive.hs b/Bustle/Noninteractive.hs
index 56e5d74..48d9461 100644
--- a/Bustle/Noninteractive.hs
+++ b/Bustle/Noninteractive.hs
@@ -25,7 +25,7 @@ where
 
 import Prelude hiding (log)
 
-import System
+import System.Exit (exitFailure)
 import System.IO (hPutStrLn, stderr)
 import Data.Maybe (fromMaybe, mapMaybe)
 import Data.List (nub)
diff --git a/Bustle/Stats.hs b/Bustle/Stats.hs
index 8a3e206..d822058 100644
--- a/Bustle/Stats.hs
+++ b/Bustle/Stats.hs
@@ -68,7 +68,7 @@ frequencies = reverse
     where alt Nothing  = Just 1
           alt (Just n) = Just (n + 1)
 
-mean :: Fractional a => [a] -> a
+mean :: (Eq a, Fractional a) => [a] -> a
 mean = acc 0 0
    where acc 0 _ [] = error "mean of empty list"
          acc n t [] = t / n
diff --git a/bustle.cabal b/bustle.cabal
index dacdf1e..c7d906e 100644
--- a/bustle.cabal
+++ b/bustle.cabal
@@ -94,7 +94,6 @@
                  , glade
                  , glib
                  , gtk > 0.11
-                 , haskell98
                  , mtl
                  , pango
                  , parsec
--
cgit v0.9.0.2-2-gbebe
